### PR TITLE
Make GCS OAuth token thread-safe

### PR DESCRIPTION
**Pull request by huanghaoyuanhhy**
issue: #22706

### CODE REVIEW

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by huanghaoyuanhhy** [comment]
rerun ut

**Comment by haorenfsa** [comment]
rerun ut

**Comment by Bennu-Li** [comment]
rerun ut

File: internal/storage/gcp/gcp.go [hunk 1]
```diff
@@ -41,14 +42,18 @@ func NewWrapHTTPTransport(secure bool) (*WrapHTTPTransport, error) {
 func (t *WrapHTTPTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// here Valid() means the token won't be expired in 10 sec
 	// so the http client timeout shouldn't be longer, or we need to change the default `expiryDelta` time
-	if !t.currentToken.Valid() {
-		var err error
-		t.currentToken, err = t.tokenSrc.Token()
+	currentToken := t.currentToken.Load()
+	if currentToken.Valid() {
+		req.Header.Set("Authorization", "Bearer "+currentToken.AccessToken)
```
Comment (congqixia) [inline internal/storage/gcp/gcp.go:47]: Maybe use `fmt.Sprintf` instead of string concat

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**